convoy/fleet.py

```diff
@@ -1677,6 +1679,23 @@ def _construct_pool_object(
                 value=block_for_gr,
             )
         )
+    # add custom env vars to the batch start task
+    if util.is_not_empty(
+            pool_settings.environment_variables_keyvault_secret_id):
+        env_vars = keyvault.get_secret(
```

Comment:
Prior to `keyvault.get_secret` it will be helpful to invoke: `_check_keyvault_client(keyvault_client)`

Reply:
Done